src/tiled/createmultipointobjecttool.cpp

```diff
@@ -75,15 +82,24 @@ void CreateMultipointObjectTool::mousePressedWhileCreatingObject(QGraphicsSceneM
         QPolygonF next = mOverlayPolygonObject->polygon();
 
         // If the last position is still the same, ignore the click
-        if (next.last() == current.last())
+        if (next.last() == current.last() && !mExtending)
```

Comment:
Why is this check irrelevant while extending? I think it was introduced to avoid inserting overlapping points in case the user double-clicks. It seems to me like it would be useful to check the first or last point, depending on `mExtendingFirst`.